Composer/packages/client/src/pages/design/DebugPanel/TabExtensions/WebchatLog/WebchatLogContent.tsx

```diff
@@ -2,29 +2,97 @@
 // Licensed under the MIT License.
 
 /** @jsx jsx */
-import { jsx } from '@emotion/core';
-import { useMemo, useEffect, useState, useRef } from 'react';
+import { css, jsx } from '@emotion/core';
+import React, { useMemo, useEffect, useState, useRef, useCallback } from 'react';
 import { useRecoilValue } from 'recoil';
+import {
+  ConversationTrafficItem,
+  ConversationActivityTrafficItem,
+  ConversationNetworkTrafficItem,
+} from '@botframework-composer/types/src';
+import { NeutralColors, SharedColors } from '@uifabric/fluent-theme';
+import { ConversationNetworkErrorItem } from '@botframework-composer/types';
 
-import { rootBotProjectIdSelector } from '../../../../../recoilModel';
-import { webChatLogsState } from '../../../../../recoilModel/atoms';
+import {
+  dispatcherState,
+  rootBotProjectIdSelector,
+  webChatTraffic,
+  webChatInspectionData,
+} from '../../../../../recoilModel';
 import { DebugPanelTabHeaderProps } from '../types';
+import { WebChatInspectionData } from '../../../../../recoilModel/types';
+import { getDefaultFontSettings } from '../../../../../recoilModel/utils/fontUtil';
 
-import { WebchatLogItem } from './WebchatLogItem';
+import { WebChatInspectorPane } from './WebChatInspectorPane';
+
+const isActive = false; // TODO: put this into a component
+const hoverItem = css`
+  padding: 0 16px;
+  background-color: ${isActive ? NeutralColors.gray30 : NeutralColors.white};
+  &:hover {
+    background-color: ${isActive ? NeutralColors.gray40 : NeutralColors.gray20};
+  }
+`;
+
+const pointer = css`
+  cursor: pointer;
+`;
+
+const link = css`
+  text-decoration: underline;
+  color: ${SharedColors.blue10};
+  display: inline-block;
+  padding-right: 4px;
+`;
+
+const DEFAULT_FONT_SETTINGS = getDefaultFontSettings();
+const logItem = css`
+  fontSize: ${DEFAULT_FONT_SETTINGS.fontSize},
+  fontFamily: ${DEFAULT_FONT_SETTINGS.fontFamily},
+`;
+
+const networkItem = css`
+  display: flex;
+  flex-direction: column;
+`;
+
+const emphasizedText = css`
+  color: ${NeutralColors.black};
+`;
+
+const redErrorText = css`
+  color: ${SharedColors.red20};
+`;
+
+const timestampStyle = css`
+  color: ${SharedColors.green20};
+  padding-right: 6px;
+`;
+
+const timestampBracket = css`
+  color: ${NeutralColors.gray130};
+`;
+
+const emptyStateMessage = css`
+  padding-left: 16px;
+`;
+
+const getActivityDirection = (activity) => {
+  if (activity?.recipient && activity.recipient.role === 'bot') {
+    return <span>{'->'}</span>;
```

Comment:
nit: dont need { and }

Reply:
it's either `{'->'}` or `-&gt;` -- I think the former is more readable

Reply:
What if we just use &rarr; and get an actual right-arrow?